File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    par = dict()
    for p in participant:
        if p in par.keys():
            par[p] += 1
        else:
            par[p] = 1
            
    for c in completion:
        par[c] -= 1
            
    answer = [p for p in par if par[p] == 1][0]
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        List<String> pList = new ArrayList<>(Arrays.asList(participant));
        List<String> cList = new ArrayList<>(Arrays.asList(completion));
        HashMap<String, Integer> map = new HashMap<>();
        
        for (String p : pList) {
            if (map.containsKey(p)) {
                map.put(p, map.get(p)+1);
            } else {
                map.put(p, 1);
            }
        }
        
        for (String c : cList) {
            map.put(c, map.get(c)-1);
        }
        
        for (String k : map.keySet()) {
            if (map.get(k) == 1) {
                answer = k;
            }
        }
        return answer;
    }
}